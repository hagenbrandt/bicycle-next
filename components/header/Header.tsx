export type Headline = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'

type HeaderProps = {
  headerText: string
  headerType: Headline
}

const Header = ({ headerText, headerType }: HeaderProps) => {
  if (!headerText || !headerType) {
    return <></>
  }

  const headlineWithText = {
    h1: <h1>{headerText}</h1>,
    h2: <h2>{headerText}</h2>,
    h3: <h3>{headerText}</h3>,
    h4: <h4>{headerText}</h4>,
    h5: <h5>{headerText}</h5>,
    h6: <h6>{headerText}</h6>,
  }

  return headlineWithText[headerType] || headlineWithText['h1']
}

export default Header
