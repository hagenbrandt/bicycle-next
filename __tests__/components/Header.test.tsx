import { render, screen, cleanup } from '@testing-library/react'
import Header, { Headline } from '../../components/header/Header'

describe('header', () => {
  it('renders given headline tag with given text', () => {
    const headlineTypes: Headline[] = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']
    const headlineText = 'This is a headline'

    headlineTypes.forEach((item) => {
      render(<Header headerText={headlineText} headerType={item} />)

      expect(screen.getByRole('heading').tagName).toBe(item.toUpperCase())
      expect(screen.getByRole('heading')).toHaveTextContent(headlineText)

      cleanup()
    })
  })

  it('returns empty DOM element when no text is given', () => {
    const { container } = render(<Header headerText="" headerType="h1" />)

    expect(container).toBeEmptyDOMElement()
  })
})
