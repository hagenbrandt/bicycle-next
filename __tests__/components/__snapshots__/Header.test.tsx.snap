// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header makes snapshot test 1`] = `
<div>
  <h1>
    This is a headline
  </h1>
</div>
`;
